package.json

```diff
@@ -18,6 +18,7 @@
     }
   },
   "dependencies": {
+    "axios": "^0.19.0",
```

Comment:
Can we increase the abstraction of `get` `post` `delete` and `put`?
Because I know `axios` does something like
```js
axios.get(...);
axios.post(...);
```

Please consider [Request-Promise-Native](https://www.npmjs.com/package/request-promise-native)

Reply:
- I can't find a good way to set up baseURL in Request-Promise document. Then each time we make a request, we need to also import the baseURL and join the basedURL with the route. But in axios. We can just call axios.post("/signin")
- I can't find a good way to handle header in pre-request, which means you should add token to header of request if user has login, otherwise not. If I need to do it in Request-Promise, I also need to import the options each time. And the option should has a predefined header.

Reply:
- It is completely fine to import baseURL every-time when we make a request to our backend service. However why do we need to preset the base url to one and only one route? Imagine our application is a huge service, which consists service request from not one but many api gateway. Since we are accessing a different api gateway, the baseUrl will be different for all of them. Defaulting baseUrl is smart but limits our abstraction
- What is a predefined header? Not all of our APIs are secured by JWT. We don't need a predefined header every-time we access an API. Again you are limiting abstraction. Because not all headers are the same, if you are access a different api gateway, it is going to require a new/different JWT token. Predefining header is probably not a good way to do so

Reply:
I agree, if you also consider different api gateways, then axios has no benefits